dask/array/reductions.py

```diff
@@ -52,14 +60,15 @@ def reduction(
     concatenate=True,
     output_size=1,
     meta=None,
+    weights=None,
 ):
     """General version of reductions
 
     Parameters
     ----------
     x: Array
         Data being reduced along one or more axes
-    chunk: callable(x_chunk, axis, keepdims)
+    chunk: callable(x_chunk, [weights_chunk=None], axis, keepdims)
```

Comment:
It seems kind of odd to put this in the middle of the existing args.

Reply:
Hi @jsignell  - thanks for looking at this.

I understood (?) that `blockwise` assigns values from its`*args` parameter to the opening parameters of its `func`.

Given that, when we have here  `args=(x, inds, weights, inds)`, `blockwise(chunk, inds, *args, ...)` can only assign `x` and `weights` to the first and second parameters of the  `chunk` function. 

Does that sound right?